mentor-match-app/src/components/profile/LoggedUserProfile.jsx

```diff
@@ -0,0 +1,490 @@
+
+const schema = yup.object().shape({
+  name: yup.string().required('Please enter your name'),
+  title: yup.string().required('Please enter your title'),
+  affiliation: yup.string().required('Please enter your affiliation'),
+  //profilePicture: yup.string().url('Invalid URL'),
+  websiteUrl: yup.string().url('Invalid URL'),
+  publicProfile: yup.boolean()
+})
+
+const LoggedUserProfile = ({ openDialog, setOpenDialog }) => {
+  const [loading, setLoading] = useState(true)
+  const { user, refreshUser } = useUser()
+  const [formAnswers, setFormAnswers] = useState({})
+  const [enableEdit, setEnableEdit] = useState(false)
+  const { enqueueSnackbar } = useSnackbar()
+
+  const onSubmit = async (values, { setSubmitting }) => {
+    setSubmitting(true)
+    try {
+      const res = await updateUserProfile(user, values)
+      if (!res.ok) {
+        setSubmitting(false)
+        return enqueueSnackbar(res.error, { variant: 'error' })
+      }
+      enqueueSnackbar('Profile updated successfully', { variant: 'success' })
+      await refreshUser()
+      setSubmitting(false)
+      setEnableEdit(false)
+      //handleDialogOpen()
+    } catch (error) {
+      console.error('Error during profile update:', error)
+      enqueueSnackbar('An error occurred during profile update', {
+        variant: 'error'
+      })
+      setSubmitting(false)
+    }
+  }
+
+  useEffect(() => {
+    if (openDialog) {
+      const fetchFormAnswers = async () => {
+        const ans = await getFormAnswers(user.uid)
+        setFormAnswers(ans)
+        console.log('formAnswers', ans)
+        setLoading(false)
+      }
+      fetchFormAnswers()
+    }
+  }, [openDialog, user.uid])
+
+  const initialValues = {
+    name: user.displayName,
+    description: user.profileDescription,
+    affiliation: user.affiliation,
+    title: user.title,
+    location: user.location,
+    profilePicture: user.profilePicture,
+    profileDescription: user.profileDescription,
+    websiteUrl: user.websiteUrl
+  }
+
+  return (
+    <Dialog
+      open={openDialog}
+      onClose={() => setOpenDialog(false)}
+      maxWidth={'lg'}
+      fullWidth
+    >
+      <DialogContent>
+        {loading ? (
+          <Box
+            display="flex"
+            justifyContent="center"
+            alignItems="center"
+            pt={2}
+          >
+            <CircularProgress />
+          </Box>
+        ) : (
+          <Stack
+            //alignItems={'center'}
+            sx={{ width: '100%' }}
+            spacing={1}
+            pb={5}
+          >
+            <IconButton
+              onClick={() => {
+                setOpenDialog(false)
+                setEnableEdit(false)
+              }}
+              color="text.secondary"
+              sx={{ alignSelf: 'flex-end' }}
+            >
+              <CloseIcon />
+            </IconButton>
+            <Stack direction="row" spacing={2} width={'100%'} pr={6} pl={2}>
+              <Formik
```

Comment:
Optimize Formik Configuration

```suggestion
             <Formik
    initialValues={initialValues}
-   validationSchema={schema}
-   onSubmit={onSubmit}
-   sx={{ width: '100%' }}
+   validationSchema={validationSchema}
+   onSubmit={handleSubmit}
+   enableReinitialize
  >
```